mio/devices/gs/header.py

```diff
@@ -55,10 +56,11 @@ def from_buffer(
     ) -> tuple[Self, np.ndarray]:
         """Split buffer into a :class:`.GSBufferHeader` and a 1D, 16-bit pixel array."""
         header_start = len(config.preamble) * config.dummy_words
+        print(header_start)
         header_end = header_start + (header_fmt.header_length * 4)
         header_array = np.frombuffer(buffer[header_start:header_end], dtype=np.uint32)
         header = cls.from_format(header_array, header_fmt, construct=True)
-        payload = buffer_to_array(buffer[header_end:])
+        payload = buffer_to_array(buffer[header_end+1:]) # added +1 for header end+1
```

Comment:
python is zero-indexed, this is incorrect. try testing your assumptions in reduced examples (or writing tests!!!!!!!)

```python
>>> mylist = (["preamble"] * 3) + (["header"] * 5) + (["not header"] * 5)
>>> mylist
['preamble',
 'preamble',
 'preamble',
 'header',
 'header',
 'header',
 'header',
 'header',
 'not header',
 'not header',
 'not header',
 'not header',
 'not header']
>>> header_len = 5
>>> header_start = 3
>>> header_end = header_start + header_len
>>> mylist[header_start:header_end]
['header', 'header', 'header', 'header', 'header']
>>> mylist[header_end:]
['not header', 'not header', 'not header', 'not header', 'not header']
```